spyder/api/plugins/new_api.py

```diff
@@ -410,43 +614,98 @@ def _unregister(self):
         self.is_compatible = None
         self.is_registered = False
 
+    # ---- Convinience attributes as properties
+    # -------------------------------------------------------------------------
+    @property
+    def main(self) -> spyder.app.mainwindow.MainWindow:
+        """Spyder main window to which this plugin belongs; i.e. its parent."""
+        return self._main
+
+    @main.setter
+    def main(self, value: spyder.app.mainwindow.MainWindow) -> None:
+        """Spyder main window to which this plugin belongs; i.e. its parent."""
+        self._main = value
+
     # ---- API: available methods
     # -------------------------------------------------------------------------
-    def get_path(self):
+    def get_path(self) -> str:
         """
-        Return the plugin's system path.
+        Return the path on disk to the plugin's root module directory.
+
+        Returns
+        -------
+        str
+            The path to the directory in which the plugin's ':file:`plugin.py`
+            module is located, i.e. its top-level site package directory.
         """
         return self._plugin_path
 
-    def get_container(self):
+    def get_container(self) -> PluginMainContainer | None:
```

Comment:
This returns `PluginMainWidget` for dockable plugins, but I don't know if it's worth it to document that.